LanguageFeatures/primary-constructors/static_processing_A05_t01.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A compile-time error occurs if the name of the primary
+/// constructor is the same as the name of a constructor (declaring or not) in
+/// the body.
+///
+/// @description Check that it is a compile-time error if the name of the
+/// primary constructor is the same as the name of some constructor declared in
+/// the body. Test classes.
+/// @author sgrekhov22@gmail.com
+
+// TODO (sgrekhov) Add `declaring-constructors` experimental flag
+
+class C1(var int v) {
+  C1(this.v);
+//^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+class C2 {
+  this(var int v);
+  C2(int v);
+//^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+class const C3.someName(final int v) {
+  const C3.someName(this.v);
+//      ^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+class C4 {
+  this.someName(var int v);
+  C4.someName(int v);
```

Comment:
Again, no primary constructors here, but the name clash is fine!

Reply:
Again changed to have the primary constructor.